### PR TITLE
ufs_public_release: updates from DTC and GSD, namelists for GFSv15.2 and GFSv16beta, update FMS submodule pointer

### PR DESCRIPTION
**Pull request by climbfuji**
This PR contains:

- update FMS to point to tag 2019.01 of NOAA-GFDL
- update of path to CCPP physics library for CCPP regression tests
- add -Wall to compiler flags for GNU compilers
- remove warnings for non-existent include directories for GNU compiler
- update of regression testing scripts to detect errors in tests/run_test.sh
- new regression tests fv3_ccpp_gfs_v15p2, fv3_ccpp_gfs_v15p2_debug, fv3_ccpp_gfs_v16beta, fv3_ccpp_gfs_v16beta_debug
- update of rt.conf to run only those tests, update of RTPWD in rt.sh
- make compile_cmake.sh work with Cheyenne Intel/GNU, and bugfixes for compile_cmake.sh

### CODE REVIEW

**Comment by climbfuji** [comment]
Associated PRs:
https://github.com/noaa-psd/stochastic_physics/pull/7
https://github.com/NCAR/ccpp-framework/pull/245
https://github.com/NCAR/ccpp-physics/pull/369
https://github.com/NOAA-EMC/fv3atm/pull/22
https://github.com/ufs-community/ufs-weather-model/pull/15

For regression testing, see below.

**Comment by climbfuji** [comment]
Manual testing on macOS+GNU: code compiles and gives b4b identical results between IPD, CCPP dynamic and CCPP static in REPRO mode for fv3_gfdlmp / fv3_ccpp_gfdlmp.

**Comment by climbfuji** [comment]
Regression testing on hera/intel: will start after hera comes out of maintenance.

**Comment by climbfuji** [comment]
Follow-up commits to this PR will modify the regression test config `rt.conf` to only include two (new) tests on hera.intel, cheyenne.intel, cheyenne.gnu. Before making this switch, I ran the full set of regression tests (original `rt.conf` and `NEMSCompsetRun`) on hera.intel.

Original `rt.conf`: all tests pass except
- fv3_gfdlmprad_noahmp (IPD) in PROD, DEBUG
- fv3_ccpp_gfdlmprad_noahmp (CCPP) in PROD, DEBUG
- fv3_ccpp_gsd (CCPP) in PROD, DEBUG

This is expected, because the updates to the GSD suite and the bugfix for the NoahMP precipitation rates are changing the answer.

[rt_full.log](https://github.com/ufs-community/ufs-weather-model/files/3952265/rt_full.log)
[rt_132_fv3_ccpp_gsd_prod.log](https://github.com/ufs-community/ufs-weather-model/files/3952266/rt_132_fv3_ccpp_gsd_prod.log)
[rt_124_fv3_ccpp_gfdlmprad_noahmp_prod.log](https://github.com/ufs-community/ufs-weather-model/files/3952267/rt_124_fv3_ccpp_gfdlmprad_noahmp_prod.log)
[rt_099_fv3_ccpp_gsd_repro.log](https://github.com/ufs-community/ufs-weather-model/files/3952268/rt_099_fv3_ccpp_gsd_repro.log)
[rt_091_fv3_ccpp_gfdlmprad_noahmp_repro.log](https://github.com/ufs-community/ufs-weather-model/files/3952269/rt_091_fv3_ccpp_gfdlmprad_noahmp_repro.log)
[rt_051_fv3_gfdlmprad_noahmp_repro.log](https://github.com/ufs-community/ufs-weather-model/files/3952270/rt_051_fv3_gfdlmprad_noahmp_repro.log)
[rt_009_fv3_gfdlmprad_noahmp.log](https://github.com/ufs-community/ufs-weather-model/files/3952271/rt_009_fv3_gfdlmprad_noahmp.log)

`NEMSCompsetRun`: all tests pass except fv3_gfdlmprad_noahmp for the same reason as above.

[NEMSCompsetRun.log](https://github.com/ufs-community/ufs-weather-model/files/3952275/NEMSCompsetRun.log)

Update 2019/12/14: I reran the `rt.conf` tests after making the code changes (reducing the size of the commit as discussed with Jun and Arun) and after cleaning up the commit history. The result is the same, i.e. the same tests failed as described above. OK.

**Comment by climbfuji** [comment]
Regression testing using new rt.conf on Cheyenne with Intel GNU successful - first create new baseline, then rsync to final RTPWD, then test against RTPWD. Regression test logs have been updated in the repository.

[rt_gnu_create.log](https://github.com/ufs-community/ufs-weather-model/files/3957617/rt_gnu_create.log)
[rt_gnu_verify.log](https://github.com/ufs-community/ufs-weather-model/files/3957618/rt_gnu_verify.log)
[rt_intel_create.log](https://github.com/ufs-community/ufs-weather-model/files/3957619/rt_intel_create.log)
[rt_intel_verify.log](https://github.com/ufs-community/ufs-weather-model/files/3957620/rt_intel_verify.log)

File: modulefiles/cheyenne.gnu/fv3 [hunk 1]
```diff
@@ -37,3 +37,11 @@ module load NCEPlibs/9.9.9
 ## SIONlib library
 ##
 module load SIONlib/1.7.4
+
+##
+## set cmake compielrs
+##
```
Comment (DusanJovic-NOAA) [inline modulefiles/cheyenne.gnu/fv3:43]: compilers

Comment (climbfuji) [reply modulefiles/cheyenne.gnu/fv3:43]: Thanks, fixed it in both modulefiles/cheyenne.gnu/fv3 and modulefiles/cheyenne.intel/fv3.

**Comment by climbfuji** [comment]
@junwang-noaa @DusanJovic-NOAA @arunchawla-NOAA @ligiabernardet I fixed the .gitmodules typo that Dusan mentioned. I also updated the pull requests based on the conversation we had earlier this morning to minimize the changes in fv3atm and ufs-weather-model. I reran the regression tests for the two new tests and updated the regression test logs against the new baseline directory ufs-public-release-20191212 (on hera.intel, cheyenne.intel, cheyenne.gnu). I am also rerunning this against the previous baseline trunk-20191021 to make sure that the differences I get are the same than before the latest round of updates (i.e. only NoahMP and GSD physics) - just to make sure that everything is correct. Thanks for your patience!

**Review by tanyasmirnova — APPROVED** [review]

**Review by DusanJovic-NOAA — APPROVED** [review]

**Comment by DusanJovic-NOAA** [comment]
Default ccpp suite in build.sh is set to FV3_GFS_2017_gfdlmp (CCPP_SUITES variable). SDF for FV3_GFS_2017_gfdlmp is removed from fv3atm/ccpp/suties. Which suite should be new default?

**Comment by climbfuji** [comment]
Only two to choose from! I would say GFSv15p2, but I would be ok with GFSv16beta as well. Jun, Arun?

**Pull request merged**